calyx-py/calyx/gen_exp.py

```diff
@@ -85,7 +86,10 @@ def generate_fp_pow_component(
 
     comp.control += [
         init,
-        while_(lt.out, cond, par(execute_mul, incr_count)),
+        while_with(
+            CellGroupBuilder(lt, cond),
```

Comment:
Purely for ease of reading purposes (and in a future PR), maybe in the future we can define a `builder.cell_with_cb` method or something so we don't have to write out `CellGroupBuilder` which feels a bit verbose. Also, does the class expose any methods for building things? If not, the name might be simplified a bit more.

Reply:
I've renamed it to `CellAndGroup`, which is... earnest, I guess, but not any less verbose haha. 

In reality I think people should just use the helpers to get the cell and the group in one go, as I do in the FIFO and PIFO programs. Then there won't be any need to write out the name of the class. 

It would spawn more cells -- one per group, without an opportunity to reuse the same primitive cell for multiple combinational groups -- but that should be compiled away anyway, no?

To give a concrete example, `gen_msb.py` does:
```
neq = comp.cell("neq", Stdlib.op("neq", width, signed=False))

with comp.comb_group("cur_val_cond") as cur_val_cond:
        neq.left = const(width, 0)
        neq.right = cur_val.out

with comp.comb_group("count_cond") as count_cond:
        neq.left = const(width, 0)
        neq.right = counter.out

while_with(
            CellAndGroup(neq, cur_val_cond),
            [incr_count, shift_cur_val],
        ),
...        
while_with(
            CellAndGroup(neq, count_cond),
            [decr_count, shift_val_build],
        ),
```

And I am proposing that it should instead do:
```

cur_val = comp.neq_use(const(width, 0), cur_val.out, width)
count = comp.neq_use(const(width, 0), counter.out, width)

while_with(
            cur_val,
            [incr_count, shift_cur_val],
        ),
...        
while_with(
            count,
            [decr_count, shift_val_build],
        ),

Reply:
Where, in my proposal, `cur_val` and `count` are now tuples (technically objects of the class `CellAndGroup`), which we don't unwrap and rewrap. They have their own `neq` cells, which is luxurious and in this case wasteful. But let's realize that fact later and compile it away?